File: Src/CSharpier.Rider/src/main/java/com/intellij/csharpier/NullCSharpierProcess.java
package com.intellij.csharpier;

public class NullCSharpierProcess implements ICSharpierProcess {

    public static ICSharpierProcess Instance = new NullCSharpierProcess();

    private NullCSharpierProcess() {}

    @Override
    public String getVersion() {
        return "NULL";
    }

    @Override
    public boolean getProcessFailedToStart() {
        return false;
    }

    @Override
    public String formatFile(String content, String fileName) {
        return "";
    }

    @Override
    public void dispose() {}
}

File: Src/CSharpier.VSCode/src/NullCSharpierProcess.ts
import { Logger } from "./Logger";
import { ICSharpierProcess } from "./ICSharpierProcess";

export class NullCSharpierProcess implements ICSharpierProcess {
    static instance: NullCSharpierProcess;
    private logger: Logger;

    static create(logger: Logger) {
        this.instance = new NullCSharpierProcess(logger);
    }

    private constructor(logger: Logger) {
        this.logger = logger;
    }

    getProcessFailedToStart(): boolean {
        return false;
    }

    formatFile(content: string, filePath: string): Promise<string> {
        this.logger.debug(
            "Skipping formatting because this is a NullCSharpierProcess. This generally indicates there was a problem starting the CSharpier process",
        );
        return Promise.resolve("");
    }

    dispose() {}

    getVersion(): string {
        return "NULL";
    }
}
